### ISSUE TITLE
error "core Unknown command(s): cmd"

### ISSUE THREAD

**Issue by ronsalas** [mickem/nscp#422]
what im doing bad.

I am executing pasive monitoring throw NSCA with NSClient++ 0.5.0.62 2016-09-14 x64

command "nscp.exe test"

```
C:\Program Files\NSClient++>nscp.exe test
L     client Module: CommandClient
L     client Command:
L     client Extra Query:
L     client Mode: 3
L     client Boot: 1
L     client Load All: 0
L     client Warning module and boot specified only THAT module will be loaded
L     client Arguments:
D       core NSClient++ 0.5.0.62 2016-09-14 x64 Loading settings and logger...
D       core Settings not ready so we cant lookup: base-path
D       core Settings not ready so we cant lookup: exe-path
D   settings Boot.ini found in: C:\Program Files\NSClient++/boot.ini
D       core Settings not ready so we cant lookup: shared-path
D   settings Activating: ini://${shared-path}/nsclient.ini
D   settings Creating instance for: ini://${shared-path}/nsclient.ini
D       core Settings not ready so we cant lookup: shared-path
D   settings Loading: C:\Program Files\NSClient++/nsclient.ini
D       core NSClient++ 0.5.0.62 2016-09-14 x64 booting...
D       core Booted settings subsystem...
D       core On crash: restart: NSCP
D       core Archiving crash dumps in: C:\Program Files\NSClient++/crash-dumps
D       core booting::loading plugins
D       core Found: CheckDisk
D       core Found: CheckEventLog
D       core Found: CheckExternalScripts
D       core Found: CheckHelpers
D       core Found: CheckSystem
D       core Found: CheckTaskSched
D       core Found: CheckWMI
D       core Found: CommandClient
D       core Found: NSCAClient
D       core Found: Scheduler
D       core C:\Program Files\NSClient++/modules\CheckDisk.dll.dll
D       core adding C:\Program Files\NSClient++/modules\CheckDisk.dll
D       core C:\Program Files\NSClient++/modules\CheckEventLog.dll.dll
D       core adding C:\Program Files\NSClient++/modules\CheckEventLog.dll
D       core C:\Program Files\NSClient++/modules\CheckExternalScripts.dll.dll
D       core adding C:\Program Files\NSClient++/modules\CheckExternalScripts.dll

D       core C:\Program Files\NSClient++/modules\CheckHelpers.dll.dll
D       core adding C:\Program Files\NSClient++/modules\CheckHelpers.dll
D       core C:\Program Files\NSClient++/modules\CheckSystem.dll.dll
D       core adding C:\Program Files\NSClient++/modules\CheckSystem.dll
D       core C:\Program Files\NSClient++/modules\CheckTaskSched.dll.dll
D       core adding C:\Program Files\NSClient++/modules\CheckTaskSched.dll
D       core C:\Program Files\NSClient++/modules\CheckWMI.dll.dll
D       core adding C:\Program Files\NSClient++/modules\CheckWMI.dll
D       core C:\Program Files\NSClient++/modules\CommandClient.dll.dll
D       core adding C:\Program Files\NSClient++/modules\CommandClient.dll
D       core C:\Program Files\NSClient++/modules\NSCAClient.dll.dll
D       core adding C:\Program Files\NSClient++/modules\NSCAClient.dll
D       core C:\Program Files\NSClient++/modules\Scheduler.dll.dll
D       core adding C:\Program Files\NSClient++/modules\Scheduler.dll
D       core Loading plugin: CheckDisk
D       core Loading plugin: CheckEventLog
D       core Loading plugin: CheckExternalScripts
D       core Loading plugin: CheckHelpers
D       core Loading plugin: CheckSystem
D       core Loading plugin: CheckTaskSched
D   schedule Adding scheduled item: Check Apps[1684611180] = {tpl: {alias: Check
 Apps, path: /settings/scheduler/schedules/Check Apps, is_tpl: false, parent: de
fault, value: alias_apps, options : { } }, command: alias_apps, channel: NSCA, s
ource_id: , target_id: , duration: 60s}
D       core NSClient++ - 0.5.0.62 2016-09-14 Started!
D       core Found module: CommandClient...
D        cli Enter command to execute, help for help or exit to exit...
D  w32system Loading counter: disk_queue_length_0 C: Q: = \\Host\Disco fí
sico(0 C: Q:)\% de tiempo de disco
D  w32system Loading counter: disk_queue_length__Total = \\Host\Disco fís
ico(_Total)\% de tiempo de disco
E       core Unknown command(s): cmd available commands: commands {, alias_apps,
 check_always_critical, check_always_ok, check_always_warning, check_and_forward
, check_counter, check_cpu, check_critical, check_drivesize, check_eventlog, che
ck_files, check_memory, check_multi, check_negate, check_network, check_ok, chec
k_os_version, check_pagefile, check_pdh, check_process, check_service, check_tas
ksched, check_timeout, check_uptime, check_version, check_warning, check_wmi, ch
eckalwayscritical, checkalwaysok, checkalwayswarning, checkcounter, checkcpu, ch
eckcritical, checkdrivesize, checkeventlog, checkfiles, checkmem, checkmultiple,
 checkok, checkprocstate, checkservicestate, checktasksched, checkuptime, checkv
ersion, checkwarning, checkwmi, filter_perf, negate, render_perf, submit_nsca, t
imeout, xform_perf}, plugins {, 0, 1, 2, 3, 4, 5, 6, 8}
                    c:\source\master\service\NSClient++.cpp:908
L        cli Ctrl+c is not exit...
D       core Attempting to stop all plugins
D       core Stopping all plugins
D       core Unloading plugin: CheckDisk...
D       core Unloading plugin: CheckEventLog...
D       core Unloading plugin: CheckExternalScripts...
D       core Unloading plugin: CheckHelpers...
D       core Unloading plugin: CheckSystem...
D       core Unloading plugin: CheckTaskSched...
D       core Unloading plugin: CheckWMI...
D       core Unloading plugin: CommandClient...
D       core Unloading plugin: NSCAClient...
D       core Unloading plugin: Scheduler...
D       core Stopping: COM helper
D       core Stopping: Settings instance
Done
```

file nsclient.ini

```
; [NSCA Agent]

[/modules]
CheckWMI=enabled
CheckSystem=enabled
CheckDisk=enabled
CheckTaskSched=enabled
CheckExternalScripts=enabled
CheckHelpers=enabled
CheckEventLog=enabled
Scheduler=enabled
NSCAClient=enabled
CommandClient = 1

[/settings/external scripts/wrappings]

bat = scripts\\%SCRIPT% %ARGS%
ps1 = cmd /c echo scripts\\%SCRIPT% %ARGS%; exit($lastexitcode) | powershell.exe -command -
vbs = cscript.exe //T:90 //NoLogo scripts\\lib\\wrapper.vbs %SCRIPT% %ARGS%
exe = cmd /c %SCRIPT% %ARGS%



[/settings/external scripts/alias]

alias_apps = cmd /c echo scripts\\check_windows_app_installs.ps1; exit($lastexitcode) | powershell.exe -command -

[/settings/scheduler/schedules/default]
interval=1m

[/settings/scheduler/schedules]
Check Apps = alias_apps


[/settings/NSCA/client]
hostname=Prueba12

[/settings/NSCA/client/targets/default]
address=Serverip
encryption=aes
password=secret
```

I´m follow this tutorial 

https://exchange.nagios.org/directory/Plugins/Operating-Systems/Windows/check_windows_app_installs/details

Nagios log

![imagen](https://cloud.githubusercontent.com/assets/3812957/26335900/38d8d402-3f2c-11e7-9a6d-6f69a5c2ecf4.png)

**Comment by mickem** [comment]
There were some fixes in 0.5.1 which fixed setting hostname over NSCA so please try with that recent version of 0.5.1.

**Comment by mickem** [comment]
Please reopen if you have this issue with the latest version
